tests/queries/0_stateless/02518_rewrite_aggregate_function_with_if.sql

```diff
@@ -0,0 +1,41 @@
+set optimize_rewrite_aggregate_function_with_if = false;
+
+explain syntax select sum(if(number % 2, number, 0)) from numbers(100);
+explain syntax select sum(if(number % 2, 0, number)) from numbers(100);
+
+explain syntax select sum(if(number % 2, number, null)) from numbers(100);
+explain syntax select sum(if(number % 2, null, number)) from numbers(100);
+
+explain syntax select avg(if(number % 2, number, null)) from numbers(100);
+explain syntax select avg(if(number % 2, null, number)) from numbers(100);
+
+explain syntax select quantiles(0.5, 0.9, 0.99)(if(number % 2, number, null)) from numbers(100);
+explain syntax select quantiles(0.5, 0.9, 0.99)(if(number % 2, null, number)) from numbers(100);
+
+set optimize_rewrite_aggregate_function_with_if = true;
+
+explain syntax select sum(if(number % 2, number, 0)) from numbers(100);
+explain syntax select sum(if(number % 2, 0, number)) from numbers(100);
+
+explain syntax select sum(if(number % 2, number, null)) from numbers(100);
+explain syntax select sum(if(number % 2, null, number)) from numbers(100);
+
+explain syntax select avg(if(number % 2, number, null)) from numbers(100);
+explain syntax select avg(if(number % 2, null, number)) from numbers(100);
+
+explain syntax select quantiles(0.5, 0.9, 0.99)(if(number % 2, number, null)) from numbers(100);
+explain syntax select quantiles(0.5, 0.9, 0.99)(if(number % 2, null, number)) from numbers(100);
+
+set allow_experimental_analyzer = true;
+
+explain syntax select sum(if(number % 2, number, 0)) from numbers(100);
```

Comment:
`explain syntax` doesn't show the changes, it's visible in `EXPLAIN QUERY TREE run_passes = 1 SELECT ...`